File: 刷题/Hot 100/102. 二叉树的层序遍历.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) [][]int {
	ans := [][]int{}
	if root == nil {
		return ans
	}
	qu := []*TreeNode{root}
	for i := 0; len(qu) > 0; i++ {
		ans = append(ans, []int{})
		p := []*TreeNode{}
		for j := 0; j < len(qu); j++ {
			node := qu[j]
			ans[i] = append(ans[i], node.Val)
			if node.Left != nil {
				p = append(p, node.Left)
			}
			if node.Right != nil {
				p = append(p, node.Right)
			}
		}
		qu = p
	}
	return ans
}

File: 刷题/Hot 100/102. 二叉树的层序遍历.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        if(!root){
            return ans;
        }
        queue<TreeNode*> qu;
        qu.push(root);
        while(!qu.empty()){
            int levelSize=qu.size();
            ans.push_back(vector<int>());
            for(int i=1;i<=levelSize;i++){
                auto node=qu.front();
                qu.pop();
                ans.back().push_back(node->val);
                if(node->left) qu.push(node->left);
                if(node->right) qu.push(node->right);
            }
        }
        return ans;
    }
};